nym-vpn-apple/Services/Sources/Services/NymLogger/FileLogHandler.swift

```diff
@@ -63,10 +63,24 @@ public class FileLogHandler: LogHandler {
         let data = Data(logLine.utf8)
 
         ioQueue.async {
-            if self.fileHandle == nil, let logFileURL = FileLogHandler.logFileURL {
-                self.fileHandle = try? FileHandle(forWritingTo: logFileURL)
+            do {
+                if self.fileHandle == nil, let logFileURL = FileLogHandler.logFileURL {
+                    self.fileHandle = try FileHandle(forWritingTo: logFileURL)
+                }
+                try self.fileHandle?.write(contentsOf: data)
+            } catch {
+                // The file “Logs.log” doesn’t exist.
+                let nsError = error as NSError
```

Comment:
Perhaps match against `CocoaError.fileNoSuchFile`

Reply:
Alternatively just do `FileManager.default.createFile(..)` before `self.fileHandle = try FileHandle(forWritingTo: logFileURL)`, it shouldn't do anything if file exists I think.

Reply:
> Alternatively just do `FileManager.default.createFile(..)` before `self.fileHandle = try FileHandle(forWritingTo: logFileURL)`, it shouldn't do anything if file exists I think.

If a file already exists at path, this method overwrites the contents of that file if the current process has the appropriate privileges to do so.

https://developer.apple.com/documentation/foundation/filemanager/1410695-createfile

Reply:
Uh ok. Then check if it exists I guess.